### PR TITLE
decoder impl

### PR DESCRIPTION
**Pull request by rigidus**

### CODE REVIEW

File: internal/wssubscriber/server/client.go
Comment (argishtiPub) [inline internal/wssubscriber/server/client.go:1]: why do we have a diff in wssubscriber? Does this come from an incorrect merge?
If no, please fix style checks.

File: internal/wssubscriber/source/logbroadcaster.go
Comment (argishtiPub) [inline internal/wssubscriber/source/logbroadcaster.go:1]: the same here

File: internal/indexer/collector.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ func (c *SolanaTxMetaCollector) IsFinalized() bool {
 // requestTxMetaList gets solana signatures and requests for each signature tx receipt from solana
 // and adds info to txMetaDict
 func (c *SolanaTxMetaCollector) requestTxMetaList(sigSlotList ...SolTxSigSlotInfo) error {
-	sigList := make([]solana2.Signature, 0, len(sigSlotList))
+	var sigList []solana2.Signature
```
Comment (argishtiPub) [inline internal/indexer/collector.go:57]: why? if we know the length of the slice, better to allocate memory.

File: internal/indexer/common.go [hunk 2]
```diff
@@ -308,106 +297,3 @@ func InsertBatchImpl(indexerDB DBInterface, pCounter prometheus.Counter, data []
 	pCounter.Add(float64(len(data)))
 	return res.LastInsertId()
 }
-
-type NeonTxResultInfo struct {
```
Comment (argishtiPub) [inline internal/indexer/common.go:312]: it is moved to another file, or we do not need it? `NeonTxResultInfo `

File: pkg/utils/eth.go [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+type NeonTx struct {
+	Nonce     *big.Int
+	GasPrice  *big.Int
+	GasLimit  *big.Int
+	ToAddress []byte
+	Value     *big.Int
+	CallData  []byte
+	V         *big.Int
+	R         *big.Int
+	S         *big.Int
```
Comment (argishtiPub) [inline pkg/utils/eth.go:23]: do we really need `big.Int` here?

File: pkg/utils/eth_test.go
Comment (argishtiPub) [inline pkg/utils/eth_test.go:1]: 👍

**Review by argishtiPub — COMMENTED** [review]
I have put minor comments and suggestions. 
The tests look good.
please check my comments.

File: pkg/utils/eth.go (hunk 3 above)
Comment (rigidus) [reply pkg/utils/eth.go:23]: I checked ethereum golang implementation, there is big ints, except of nonce. I can make nonce as uint64, I think.

File: internal/indexer/common.go (hunk 2 above)
Comment (rigidus) [reply internal/indexer/common.go:312]: It is moved to indexed_objects.go

**Review by argishtiPub — APPROVED** [review]
big work. 
Looks good.

**Pull request merged**